pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -504,20 +741,79 @@ func (r *ReconcileWildFlyServer) routeForWildFly(w *wildflyv1alpha1.WildFlyServe
 	return route
 }
 
+// newSecretForCR returns an empty secret for holding the secrets merge
+func (r *ReconcileWildFlyServer) txnRecoveryJBossCliPasswordSecret(w *wildflyv1alpha1.WildFlyServer) *corev1.Secret {
+	password := generateToken(16)
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getTxnRecoverySecretName(w),
+			Namespace: w.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		StringData: map[string]string{
+			"username": wflyMgmtTxnRecoveryUserName,
+			"password": password,
+		},
+	}
+	// Set WildFlyServer instance as the owner and controller
+	controllerutil.SetControllerReference(w, secret, r.scheme)
+	return secret
+}
+
+func getWildflyServerPodStatusByName(w *wildflyv1alpha1.WildFlyServer, podName string) *wildflyv1alpha1.PodStatus {
+	i := sort.Search(len(w.Status.Pods), func(i int) bool { return podName == w.Status.Pods[i].Name })
```

Comment:
why don't you just iterate on w.Status.Pods? afaict you don't need to search in the sorted order.

Reply:
that was a safeguard ;-) I was not able to find in documentation if the pods are sorted by name in the list. If they are possibly in an arbitrary order then it's dangerous to do so. The statefulset is prescribed to remove the pods in opposite order they were created
```quote
When Pods are being deleted, they are terminated in reverse order, from {N-1..0}
```
(https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#deployment-and-scaling-guarantees)

The recovery processing needs to keep the same ordering for not processing wrong pod than the one the `StatefulSet` removes later.